src/rgw/rgw_op.cc

```diff
@@ -4533,11 +4533,27 @@ int RGWCopyObj::verify_permission()
   if (op_ret < 0) {
     return op_ret;
   }
-
+  boost::optional <Policy> dest_iam_policy = get_iam_policy_from_attr(s->cct, store, dest_attrs, dest_bucket.tenant);
```

Comment:
Auto? Seems better than having gigantic type specifiers and it will still work if we ever move to C++17 and std::optional.

Reply:
That will be a glorious day!